### PR TITLE
Integrate car_range nginx module

### PR DESCRIPTION
**Pull request by tchardin**

### CODE REVIEW

**Comment by willscott** [comment]
is this ready for review, or waiting for something, @tchardin ?

File: Dockerfile [hunk 1]
```diff
@@ -76,22 +86,32 @@ ARG CONFIG="--prefix=/etc/nginx \
...
  --with-stream_realip_module \
  --with-stream_ssl_module \
  --with-stream_ssl_preread_module \
- --add-dynamic-module=/usr/src/njs/nginx \
- --add-dynamic-module=/usr/src/ngx_brotli"
+ --add-dynamic-module=/usr/src/ngx_brotli \
+ --add-dynamic-module=/usr/src/njs/nginx"
 
 RUN echo "Cloning nginx and building $NGINX_VERSION (rev $NGINX_COMMIT from '$NGINX_BRANCH' branch)" \
- && hg clone -b $NGINX_BRANCH --rev $NGINX_COMMIT https://hg.nginx.org/nginx-quic /usr/src/nginx-$NGINX_VERSION \
```
Comment (DiegoRBaquero) [inline Dockerfile:89]: Doesn't this break quic?

File: Dockerfile [hunk 2]
```diff
@@ -76,22 +86,32 @@ ARG CONFIG="--prefix=/etc/nginx \
...
  && make install
 
+# protobuf. pinned @ v3.22.1
+RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v22.1/protoc-22.1-linux-x86_64.zip && unzip protoc-22.1-linux-x86_64.zip -d /usr/local && rm protoc-22.1-linux-x86_64.zip
+
+ENV NGINX_DIR=/usr/src/nginx-$NGINX_VERSION
+
+RUN echo "Cloning car_range $NGX_CAR_RANGE_VERSION, nginx dir: $NGINX_DIR" \
+  && mv $NGINX_DIR/src/http/v2/* $NGINX_DIR/src/http/ \
+  && git clone -b $NGX_CAR_RANGE_VERSION https://github.com/filecoin-saturn/nginx-car-range.git /usr/src/ngx_car_range \
```
Comment (DiegoRBaquero) [inline Dockerfile:111]: can this be depth 1?

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -35,6 +35,7 @@ jobs:
             SATURN_NETWORK=local
             ORCHESTRATOR_URL=http://localhost
             LOG_INGESTOR_URL=http://localhost
+            LASSIE_EXCLUDE_PROVIDERS=bafzbeigai3eoy2ccc7ybwjfz5r3rdxqrinwi4rwytly24tdbh6yk7zslrm
```
Comment (DiegoRBaquero) [inline .github/workflows/integration-tests.yml:38]: Why here?

Comment (joaosa) [reply .github/workflows/integration-tests.yml:38]: I think this was related to nginx hanging 🤔

File: Dockerfile (hunk 1 above)
Comment (tchardin) [reply Dockerfile:89]: Quic isn't turned on, ruled out as too slow IIRC

File: .github/workflows/integration-tests.yml (hunk 3 above)
Comment (tchardin) [reply .github/workflows/integration-tests.yml:38]: Something we did with @guanzo when Lassie was failing so we can pass a random CID but prob not needed after your fix

Comment (guanzo) [reply .github/workflows/integration-tests.yml:38]: yeah this isn't needed anymore, the Dockerfile passes a default provider so lassie doesn't crash on startup

File: Dockerfile [hunk 4]
```diff
@@ -58,6 +67,7 @@ ARG CONFIG="--prefix=/etc/nginx \
  --pid-path=/var/run/nginx.pid \
  --lock-path=/var/run/nginx.lock \
  --user=nginx --group=nginx \
+ --with-debug \
```
[the remark this answers is not in the document]
Comment (tchardin) [reply Dockerfile:70]: Ooops good catch!

**Review by DiegoRBaquero — APPROVED** [review]

File: scripts/integration_tests.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+# negative test case
+test_cid "Qmbfrc4cF2X4KXbHuqD593SLnR2xj6hULYTnrj65wKWaKm" "$not_blocked"
+# positive test case
+test_cid "bafybeibvcisellj6bfzbas3csvioltujjmif5jqpdw5ykvvwujtvt6up7u" "$blocked"
+# positive denylist.conf test case
+test_cid "bafybeidgnebuxvarpnw2grmkgnamu6cv6" "$blocked"
+
+# download car tooling
+curl -LO -s https://github.com/ipld/go-car/releases/download/v2.8.0/go-car_2.8.0_linux_amd64.tar.gz && tar xzf go-car_2.8.0_linux_amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (tchardin) [reply scripts/integration_tests.sh:34]: Yeah , later ;)

**Pull request merged**